driver/infoblox_ipam.go

```diff
@@ -76,10 +76,28 @@ func (ibDrv *InfobloxDriver) RequestAddress(r *ipamApi.RequestAddressRequest) (*
 	if rType, ok := r.Options["RequestAddressType"]; macAddr == "" || (ok && rType == netlabel.Gateway) {
 		logrus.Debugln("Request for Gateway IP")
 
+		l := &ibclient.NetworkViewLock{Name: network.NetviewName, ObjMgr: ibDrv.objMgr, LockEA: common.EA_DOCKER_PLUGIN_LOCK,
+			LockTimeoutEA: common.EA_DOCKER_PLUGIN_LOCK_TIME}
+		err := l.Lock()
+
+		if err != nil {
+			return  &ipamApi.RequestAddressResponse{}, fmt.Errorf("Failed to Allocate Gateway IP :%s", err)
+		}
+		defer l.UnLock(false)
+
+		if r.Address != "" {
+			ipAddr, err := ibDrv.objMgr.GetFixedAddress(network.NetviewName, network.Cidr, r.Address, "")
+			if err == nil && ipAddr != nil {
```

Comment:
I see. Thank you for the referring to the example use case!!

Couple of questions regarding the use case:

1) So, what does docker think the MAC of the gateway is? Have you tested that network connection initiated by a docker container can be routed through the gateway?

2) And there's no way to inject the gateway MAC address to docker?

Thanks again!